components/core/header/nav-bar/ScrollableNavBar.vue

```diff
@@ -0,0 +1,164 @@
+<template>
+    <div class="core-scrollableNavBar">
+        <button
+            class="core-scrollableNavBar__left-arrow"
+            @click="onLeftArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': leftClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-left']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <button
+            class="core-scrollableNavBar__right-arrow"
+            @click="onRightArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': rightClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-right']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <div
+            ref="scrollableNavBarScroll"
+            class="
+                core-scrollableNavBar__scroll core-scrollableNavBar__overflow
+            "
+        >
+            <div class="core-scrollableNavBar__slot">
+                <slot></slot>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+export default {
+    name: 'CoreScrollableNavBar',
+    props: {
+        scrollDelta: {
+            default: 1000,
+            type: Number,
+        },
+    },
+    data() {
+        return {
+            leftClickable: false,
+            rightClickable: true,
+        }
+    },
+    mounted() {
+        const el = this.$refs.scrollableNavBarScroll
+        this.leftClickable = !this.checkHorizontalScrollStart(el)
+        this.rightClickable = !this.checkHorizontalScrollEnd(el)
+    },
+    methods: {
+        checkHorizontalScrollStart(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return scrollLeft <= 0
+        },
+        checkHorizontalScrollEnd(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return el.scrollWidth - el.clientWidth - scrollLeft < 1
+        },
+        horizontalScroll(delta) {
+            const el = this.$refs.scrollableNavBarScroll
+            const newScrollLeft = el && el.scrollLeft + delta
+            if (el && newScrollLeft) {
+                const leftClickable = !this.checkHorizontalScrollStart(
+                    el,
+                    newScrollLeft,
+                )
+                const rightClickable = !this.checkHorizontalScrollEnd(
+                    el,
+                    newScrollLeft,
+                )
+                el.scrollTo({ left: newScrollLeft, behavior: 'smooth' }) // be aware this is async
+                this.leftClickable = leftClickable
+                this.rightClickable = rightClickable
+            }
```

Comment:
Similar here. Not going to block but just a friendly comment.

1. In JavaScript, early-return `if` statement style is more widely adopted. I.e.:
   ```js
   if (!el || !newScrollLeft) {
     return
   }
   // Logic here
   ```
   It avoids spaghetti if-else statements and is easier to read. Nevertheless, it's still a choice rather than a rule.
1. The `const newScrollLeft = el && el.scrollLeft + delta` still smells bad but I can't give a concrete suggestion at this point.

Reply:
2. agreed~